apps/web-twig-demo/assets/scripts/toast-dynamic.ts

```diff
@@ -16,7 +16,30 @@ const addDynamicToast = (event, containerId) => {
   console.log('Created dynamic toast with config:', config);
 };
 
-// Make it available in the global scope
+export const clearQueue = (event, containerId) => {
+  const container = document.getElementById(containerId);
+  const children = container.querySelectorAll('.ToastBar');
+
+  let cleared = 0;
+
+  children.forEach((element) => {
+    const instance = Toast.getOrCreateInstance(element);
+
+    if (instance instanceof Toast && instance?.isShown) {
+      instance?.hide();
+      cleared++;
+    }
+  });
+
+  if (cleared) {
+    console.log('Cleared toast queue of length:', cleared);
+  } else {
+    console.log('No toasts to clear.');
+  }
+};
+
+// Make functions available in the global scope
 window.addDynamicToast = addDynamicToast;
+window.clearQueue = clearQueue;
```

Comment:
What about prefixing or suffixing the global scope functions? I know that collision should not happen but maybe using

```js
window.SpiritToast = {
  addDynamic = addDynamicToast,
  clearQueue
}
```

would be better.

Reply:
I would keep it like this as well. Now it is the same in the twig and the web. And as Adam mentioned, we don't do it anywhere now.